py/desitarget/cuts.py

```diff
@@ -94,22 +94,70 @@ def isFSTD_colors(gflux=None, rflux=None, zflux=None, w1flux=None, w2flux=None,
     Returns:
         mask : boolean array, True if the object has colors like an FSTD
 
+    """
+
+    if primary is None:
+        primary = np.ones_like(gflux, dtype='?')
+    fstd = primary.copy()
+
+    # Clip to avoid warnings from negative numbers.
+    gflux = gflux.clip(0)
+    rflux = rflux.clip(0)
+    zflux = zflux.clip(0)
+    
+    # colors near BD+17
+    grcolor = 2.5 * np.log10(rflux / gflux)
+    rzcolor = 2.5 * np.log10(zflux / rflux)
+    fstd &= (grcolor - 0.32)**2 + (rzcolor - 0.13)**2 < 0.06**2
+
+    return fstd
+
+def isFSTD(gflux=None, rflux=None, zflux=None, primary=None, decam_fracflux=None,
+           objtype=None, decam_snr=None, obs_rflux=None, bright=False):
+    """Select FSTD targets using color cuts and photometric quality cuts (PSF-like
+    and fracflux).  See isFSTD_colors() for additional info.
+
+    Args:
+        gflux, rflux, zflux, w1flux, w2flux: array_like
+          The flux in nano-maggies of g, r, z, w1, and w2 bands.
+        primary: array_like or None
+          If given, the BRICK_PRIMARY column of the catalogue.
+        bright: apply magnitude cuts for "bright" conditions; otherwise, choose
+          "normal" brightness standards.
+
+    Returns:
+        mask : boolean array, True if the object has colors like an FSTD
+
     Notes:
         The full FSTD target selection also includes PSF-like and fracflux
         cuts; this function is only cuts on the colors.
 
     """
-    #----- F-type standard stars
     if primary is None:
         primary = np.ones_like(gflux, dtype='?')
     fstd = primary.copy()
 
-    #- colors near BD+17; ignore warnings about flux<=0
-    with warnings.catch_warnings():
-        warnings.simplefilter('ignore')
-        grcolor = 2.5 * np.log10(rflux / gflux)
-        rzcolor = 2.5 * np.log10(zflux / rflux)
-        fstd &= (grcolor - 0.32)**2 + (rzcolor - 0.13)**2 < 0.06**2
+    # Apply the magnitude and color cuts.
+    fstd &= isFSTD_colors(primary=primary, zflux=zflux, rflux=rflux, gflux=gflux)
+
+    # Apply type=PSF, fracflux, and S/N cuts.
+    fstd &= _psflike(objtype)
+    for j in (1, 2, 4):  # g, r, z
+        fstd &= decam_fracflux[j] < 0.04
```

Comment:
Reviewing `decam_fracflux[j] < 0.04` I found some NaN's for `decam_fracflux`, e.g.:

```
from desitarget import io
import numpy as np
file = '/global/project/projectdirs/cosmo/data/legacysurvey/dr3/sweep/3.0/sweep-170p010-180p015.fits'
objects = io.read_tractor(file)
np.where(objects['DECAM_FRACFLUX'].T != objects['DECAM_FRACFLUX'].T)
(array([1, 1, 1, ..., 4, 4, 4]), array([  15436,   15461,   15467, ..., 1508219, 1508238, 1508379]))
```

Should we catch these NaNs? I'm not sure we can guarantee that `DECAM_FRACFLUX` will always be well-defined in the short-term? Or perhaps we should assume that `DECAM_FRACFLUX` should always be well-defined? In any case, I think this should throw an error, for now...